File: 백준/Bronze/1978. 소수 찾기/소수 찾기.py
import math
def primenumber(x):
    if x==1:
        return False
    for i in range(2, int(math.sqrt(x) + 1)):	# 2부터 x의 제곱근까지의 숫자
        if x%i == 0:		# 나눠떨어지는 숫자가 있으면 소수가 아님
            return False
    return True
A, B=input(), input().split()
count=0
for i in range(len(B)):
    if primenumber(int(B[i]))==True:
        count+=1
print(count)

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function isPrime(number) {
    if (number < 2) return false;
    for (let i = 2; i <= Math.sqrt(number); i++) {
        if (number % i === 0) return false;
    }
    return true;
}

function makeCombinations(numbers) {
    const combinations = new Set();
    for (let length = 1; length <= numbers.length; length++) {
        const perms = getPermutations(numbers.split(''), length);
        perms.forEach(perm => combinations.add(parseInt(perm.join(''))));
    }
    return combinations;
}

function getPermutations(arr, length) {
    if (length === 1) return arr.map(el => [el]);
    const result = [];
    arr.forEach((fixed, index, origin) => {
        const rest = [...origin.slice(0, index), ...origin.slice(index + 1)];
        const perms = getPermutations(rest, length - 1);
        const attached = perms.map(perm => [fixed, ...perm]);
        result.push(...attached);
    });
    return result;
}

function solution(numbers) {
    const combinations = Array.from(makeCombinations(numbers));
    let count = 0;
    combinations.forEach(num => {
        if (isPrime(num)) count++;
    });
    return count;
}
